### ISSUE TITLE
Defining typed constant numbers

### ISSUE THREAD

**Issue by mvertes** [containous/yaegi#631]
The following program `sample.go` triggers an unexpected error:

```go
package main

type Kind int

const (
    None Kind = 0
    Left Kind = 1 << iota
    Right
    Both Kind = Left | Right
)

func main() {
    println(None, Left, Right, Both)
}
```

Expected result:
```console
$ go run ./sample.go
0 2 4 6
```

Got:
```console
$ yaegi ./sample.go
sample.go:7:2: constant definition loop
```

**Comment by mpl** [comment]
After some investigation, it turns out that the problem is reproducible in an even simpler situation, e.g.:

```
package main

const (
	Left = 1 << iota
)

func main() {
	println(Left)
}
```

and comes down to the fact that the AST node corresponding to iota is not untyped (whereas it probably should be), hence making the isNatural check fail, and leading to a (suppressed) error behind the scenes, such as:

illegal operand types for '<<' operator

So the fix in #632 , even though being correct, just hides the symptom, and does not actually fixes the root cause.

**Comment by mvertes** [comment]
Right. Thanks for working on it and investigate a bit more on `iota`.
